File: Reference/jfsp/source/apiHW.pl
#!/usr/bin/perl

use LWP::Simple;
use JSON qw( decode_json );
use Data::Dumper;
use strict;
use warnings;

my $stid = "WBB";
my $latestobs = "1";
my $token = "1234567890";

my $apiString = "stid=${stid}&latest_obs=${latestobs}&token=${token}";

my $json = get( "http://api.mesowest.net/stations?${apiString}" );
die "Could not get ${apiString}" unless defined $json;

my $decoded_json = decode_json( $json );

print Dumper $decoded_json;

File: Reference/jfsp/source/apiHW.rb
#!/usr/bin/ruby
#Require these two libraries
require "net/http"
require "uri"
#To easily convert the returned JSON string to a Ruby hash
#also require "rubygems" and "json"

#Specify request parameters
stid = "WBB"
latestobs = "1"
token = "YourToken"

#Construct the query string
apiString = "stid="+stid+"&latest_obs="+latestobs+"&token="+token

#Parse the API URL and get the body of the response (the JSON)
uri = URI.parse("http://api.mesowest.net/stations?"+apiString)
response = Net::HTTP.get_response(uri)
dataString = response.body #JSON comes in as a string
#data = JSON.parse(dataString) Converts JSON string to a Ruby hash
#Do stuff with the data!